src/database/hooks/useLocalDatabaseHook.js

```diff
@@ -46,8 +47,13 @@ const useLocalDatabaseHook = () => {
     const shouldDbMigrated = await LocalDatabaseMigration.shouldDbMigrated();
     let scopedMigrationStatus = migrationStatus;
 
+    if (scopedMigrationStatus === MIGRATION_STATUS.MIGRATING) {
+      return null;
+    }
+
     if (isEnteringApp && shouldDbMigrated) {
-      await LocalDatabaseMigration.migrateDb();
+      setMigrationStatus(MIGRATION_STATUS.MIGRATING);
+      if (withMigration) await LocalDatabaseMigration.migrateDb();
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The logic here checks if the database is currently migrating and if so, it returns null. Then, if the app is entering and the database should be migrated, it sets the migration status to 'MIGRATING', performs the migration if the `withMigration` flag is true, and then sets the migration status to 'MIGRATED'. This is a good approach as it ensures that the database is in a consistent state before any operations are performed on it.

```diff
+    if (scopedMigrationStatus === MIGRATION_STATUS.MIGRATING) {
+      return null;
+    }
+
+    if (isEnteringApp && shouldDbMigrated) {
+      setMigrationStatus(MIGRATION_STATUS.MIGRATING);
+      if (withMigration) await LocalDatabaseMigration.migrateDb();
+      setMigrationStatus(MIGRATION_STATUS.MIGRATED);
+      scopedMigrationStatus = MIGRATION_STATUS.MIGRATED;
+    }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->